File: src/main/java/tk/loryruta/hgame/scenario/scheduler/TaskHandle.java
package tk.loryruta.hgame.scenario.scheduler;

import lombok.Getter;

public class TaskHandle {
    @Getter
    private final int id;

    @Getter
    private final Runnable action;

    @Getter
    private final boolean repeating;

    private long delay, callAt;

    public TaskHandle(int id, Runnable action, long delay, boolean repeating) {
        this.id = id;
        this.action = action;
        this.repeating = repeating;

        this.delay = delay;
        this.callAt = System.currentTimeMillis() + delay;
    }

    public boolean isReady(long time) {
        return time >= callAt;
    }

    public void run() {
        action.run();
        if (repeating) {
            callAt = System.currentTimeMillis() + delay;
        }
    }
}

File: game/src/main/kotlin/xyz/upperlevel/hgame/world/scheduler/TaskHandle.kt
package xyz.upperlevel.hgame.world.scheduler

class TaskHandle(val id: Int,
                 val action: () -> Unit,
                 private val delay: Long,
                 val isRepeating: Boolean) {
    private var callAt: Long = 0

    init {
        this.callAt = System.currentTimeMillis() + delay
    }

    fun isReady(time: Long): Boolean {
        return time >= callAt
    }

    fun run() {
        action()
        if (isRepeating) {
            callAt = System.currentTimeMillis() + delay
        }
    }
}
